packages/crypto-wallet-core/src/validation/xrp/index.ts

```diff
@@ -0,0 +1,34 @@
+import { IValidation } from '..';
+import baseX from 'base-x';
+import crypto from 'crypto';
+
+export class XrpValidation implements IValidation {
+  validateAddress(_network: string, address: string): boolean {
+    // First ensure it matches regex
+    const RippleAddressRegex = new RegExp(/^r[rpshnaf39wBUDNEGHJKLM4PQRST7VWXYZ2bcdeCg65jkm8oFqi1tuvAxyz]{27,35}$/);
+    if (!address.match(RippleAddressRegex)) {
+      return false;
+    }
+
+    // Then ensure it is a valid base58check encoding
+    const base58 = baseX('rpshnaf39wBUDNEGHJKLM4PQRST7VWXYZ2bcdeCg65jkm8oFqi1tuvAxyz');
```

Comment:
is this the base58 character set?

If so, pull that into a variable called characterSet, otherwise I'd think this was an address

Reply:
Fixed